internal/client/client_integration_test.go

```diff
@@ -0,0 +1,87 @@
+// +build integration
+
+package client
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/newrelic/newrelic-client-go/pkg/region"
+
+	"github.com/newrelic/newrelic-cli/internal/credentials"
+)
+
+var overrideEnvVars = []string{
+	"NEW_RELIC_API_KEY",
+	"NEW_RELIC_REGION",
+}
+
+func TestApplyOverrides(t *testing.T) {
+	// Do not run this in parallel, we are messing with the environment
+
+	f, err := ioutil.TempDir("/tmp", "newrelic")
+	assert.NoError(t, err)
+	defer os.RemoveAll(f)
+
+	// Initialize the new configuration directory
+	c, err := credentials.LoadCredentials(f)
+	assert.NoError(t, err)
+
+	// Create an initial profile to work with
+	err = c.AddProfile("testCase1", "us", "apiKeyGoesHere")
+	assert.NoError(t, err)
+	p := c.Profiles["testCase1"]
+	assert.NotNil(t, p)
+
+	// Clear env vars we are going to mess with, and reset on exit
+	for _, v := range overrideEnvVars {
+		if val, ok := os.LookupEnv(v); ok {
+			defer os.Setenv(v, val)
+			os.Unsetenv(v)
+		}
+	}
+
+	// Default case (no overrides)
+	p2 := applyOverrides(&p)
+	assert.NotNil(t, p2)
+	assert.Equal(t, p.APIKey, p2.APIKey)
+	assert.Equal(t, p.Region, p2.Region)
+
+	// Override just the API Key
+	os.Setenv("NEW_RELIC_API_KEY", "anotherAPIKey")
+	p2 = applyOverrides(&p)
+	assert.NotNil(t, p2)
+	assert.Equal(t, "anotherAPIKey", p2.APIKey)
+	assert.Equal(t, p.Region, p2.Region)
+
+	// Both
+	os.Setenv("NEW_RELIC_REGION", "US")
```

Comment:
Perhaps too we should test something that isn't the US, since its the default and could give us false results.

Reply:
The goal is to check that the env overrides happen, vs testing that Region parsing is working correctly (which happens in `newrelic-client-go`).  That said, I can change one to `eu`